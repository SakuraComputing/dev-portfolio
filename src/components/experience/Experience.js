import React from 'react';
import ExperienceItem from '../../components/experience/ExperienceItem';

class Experience extends React.Component{

    render() {

        const { experience } = this.props;

        let experienceItems;

        if(experience.length > 0) {
            experienceItems = experience.map(role => (
                <ExperienceItem key={role.id} role={role}/>
            ))
        }

        return(
            <section className="experience section">
                <div className="section-inner">
                    <h2 className="heading">Work Experience</h2>
                    <div className="content">
                        {experienceItems}
                    </div>
                </div>
            </section>
        )
    }
}
export default Experience;