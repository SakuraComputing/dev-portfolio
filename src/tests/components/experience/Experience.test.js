import React from 'react';
import { shallow } from 'enzyme';
import Experience from '../../../components/experience/Experience';

describe('Experience Section', () => {
    it('should render correctly', () => {
        const experience = jest.fn();
        const wrapper = shallow(<Experience experience={experience}/>);
        expect(wrapper).toMatchSnapshot();
    });
});