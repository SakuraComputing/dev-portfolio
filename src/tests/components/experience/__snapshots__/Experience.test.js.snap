// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Experience Section should render correctly 1`] = `
<section
  className="experience section"
>
  <div
    className="section-inner"
  >
    <h2
      className="heading"
    >
      Work Experience
    </h2>
    <div
      className="content"
    />
  </div>
</section>
`;
